bundle/edu.gemini.p2checker/src/main/java/edu/gemini/p2checker/rules/general/GeneralRule.java

```diff
@@ -310,17 +312,14 @@ public IP2Problems check(final ObservationElements elements)  {
             if (elements.getTargetObsComp().isEmpty()) return null; // can't perform this check without a target environment
 
             final P2Problems problems = new P2Problems();
-            final SPTarget scienceTarget = elements.getTargetObsComp().getValue().getBase();
 
             if (!hasScienceObserves(elements.getSequence())) return null; //if there are not observes, ignore this check (SCT-260)
 
-            if (scienceTarget == null) { //really unlikely
-                problems.addError(PREFIX+"NO_SCIENCE_TARGET_MSG", NO_SCIENCE_TARGET_MSG, elements.getTargetObsComponentNode().getValue());
-            } else {
-                if ("".equals(scienceTarget.getName().trim())) {
-                    problems.addError(PREFIX+"EMPTY_TARGET_NAME_MSG", EMPTY_TARGET_NAME_MSG, elements.getTargetObsComponentNode().getValue());
-                }
+            final Asterism asterism = elements.getTargetObsComp().getValue().getAsterism();
+            if ("".equals(asterism.name().trim())) {
+                problems.addError(PREFIX+"EMPTY_TARGET_NAME_MSG", EMPTY_TARGET_NAME_MSG, elements.getTargetObsComponentNode().getValue());
```

Comment:
I think the idea is to check that all science targets have a name.  This would only flag an error if none of them have a name?